### PR TITLE
Acorn optimizer: Allow top-level await expressions

### PR DESCRIPTION
**Pull request by kleisauke**
`-pthread` + `-sEXPORT_ES6` emits a top-level await expression in the main worker file after commit 49ab23a. Ensure the Acorn optimizer is aware of this.

<details>
  <summary>Details</summary>

```
file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:3567
  var err = new SyntaxError(message);
            ^

SyntaxError: Cannot use keyword 'await' outside an async function (24:28)
  const { createRequire } = await import('module');
                            ^

    at Parser.pp$4.raise (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:3567:13)
    at Parser.pp$5.checkUnreserved (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:3465:14)
    at Parser.pp$5.parseIdent (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:3479:10)
    at Parser.pp$5.parseExprAtom (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:2849:19)
    at Parser.pp$5.parseExprSubscripts (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:2709:19)
    at Parser.pp$5.parseMaybeUnary (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:2675:17)
    at Parser.pp$5.parseExprOps (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:2602:19)
    at Parser.pp$5.parseMaybeConditional (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:2585:19)
    at Parser.pp$5.parseMaybeAssign (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:2552:19)
    at Parser.pp$8.parseVar (file:///emsdk/upstream/emscripten/node_modules/acorn/dist/acorn.mjs:1315:24) {
  pos: 836,
  loc: Position { line: 24, column: 28 },
  raisedAt: 848
}
em++: error: '/emsdk/node/16.20.0_64bit/bin/node /emsdk/upstream/emscripten/tools/acorn-optimizer.mjs /src/lib/vips-node.worker.mjs minifyWhitespace --closureFriendly --exportES6' failed (returned 1)
```
</details>

### CODE REVIEW

**Comment by sbc100** [comment]
How is it that we don't see that error in our existing tests I wonder?

File: tools/acorn-optimizer.mjs [hunk 1]
```diff
@@ -2029,6 +2029,7 @@ try {
     preserveParens: closureFriendly,
     onComment: closureFriendly ? sourceComments : undefined,
     sourceType: exportES6 ? 'module' : 'script',
+    allowAwaitOutsideFunction: exportES6,
```
Comment (sbc100) [inline tools/acorn-optimizer.mjs:2032]: I think perhaps we should just unconditionally allow this.  We don't rely on acorn to enforce these kind of rules and we don't know where the final code will be run.

Comment (kripken) [reply tools/acorn-optimizer.mjs:2032]: I agree, yeah, that seems simpler and better,
```suggestion
    allowAwaitOutsideFunction: true,
```

Comment (kleisauke) [reply tools/acorn-optimizer.mjs]: Incorporated with  commit 8191423281151961b4fe8b521206a96214a2db9a, thanks!

**Comment by kleisauke** [comment]
> How is it that we don't see that error in our existing tests I wonder?

Commit 7c9773b3f3e19b5ac798ac45b7b0c620283d66a8 ensures that these lines are covered too:
https://github.com/emscripten-core/emscripten/blob/c4d76b8d8a487de7b8c2eadd91775bbf53e3efd4/tools/link.py#L1998-L2001

**Review by sbc100 — APPROVED** [review]

**Pull request merged**